ui/Decoder.elm

```diff
@@ -28,26 +28,27 @@ buildDecoder =
 downloadDecoder : Decoder Download
 downloadDecoder =
     decode Download
-        |> required "mimetype" (nullable string)
-        |> required "size" (nullable int)
+        |> required "mimetype" string
+        |> required "size" int
         |> required "url" string
 
 
 targetDecoder : Decoder Target
 targetDecoder =
     decode Target
-        |> required "version" (nullable string)
+        |> required "version" string
         |> required "platform" string
-        |> required "channel" (nullable string)
+        |> required "channel" string
         |> required "locale" string
 
 
 sourceDecoder : Decoder Source
 sourceDecoder =
     decode Source
         |> required "product" string
-        |> required "tree" string
-        |> required "revision" (nullable string)
+        |> optional "tree" (nullable string) Nothing
+        |> optional "revision" (nullable string) Nothing
+        |> optional "repository" (nullable string) Nothing
```

Comment:
The elm-decode-pipeline needs the decoder to be of the same type as the fallback, so a `Maybe` in both case. And the only way (that I know of) to do that is to use `nullable`.